vivisect/symboliks/common.py

```diff
@@ -81,6 +81,7 @@ def __init__(self):
         self.kids          = []
         self.parents       = []
         self.cache         = {}
+        self._reduced       = False
```

Comment:
what do you say we put the onus on the thing updating the AST?  since there's currently nothing that does that except walkTree, which returns either the same or a new object.

alternately, we could simply wipe this out at the end of reduce?  that sounds really painful to "just do it" for larger AST's.  i'm combatting ASTs with millions of nodes.  many of them are the same node, however... so perhaps we simply rewalk and stop when we reach something that's already cleared.